glib/src/boxed.rs

```diff
@@ -15,31 +15,33 @@ use std::ptr;
 /// Wrapper implementations for Boxed types. See `wrapper!`.
 #[macro_export]
 macro_rules! glib_boxed_wrapper {
-    ([$($attr:meta)*] $visibility:vis $name:ident, $ffi_name:ty,
+    ([$($attr:meta)*] $visibility:vis $name:ident $(<$($generic:ident $(: $($bound:tt +)* $bound2:tt)?),+>)?, $ffi_name:ty,
      @copy $copy_arg:ident $copy_expr:expr, @free $free_arg:ident $free_expr:expr
      $(, @type_ $get_type_expr:expr)?) => {
-        $crate::glib_boxed_wrapper!(@generic_impl [$($attr)*] $visibility $name, $ffi_name);
+        $crate::glib_boxed_wrapper!(@generic_impl [$($attr)*] $visibility $name $(<$($generic $(: $($bound)* $bound2)?),+>)?, $ffi_name);
+
         $crate::glib_boxed_wrapper!(
-            @memory_manager_impl $name, $ffi_name,
+            @memory_manager_impl $name $(<$($generic $(: $($bound)* $bound2)?),+>)?, $ffi_name,
             @copy $copy_arg $copy_expr, @free $free_arg $free_expr
         );
-        $($crate::glib_boxed_wrapper!(@value_impl $name, $ffi_name, @type_ $get_type_expr);)?
+
+        $crate::glib_boxed_wrapper!(@value_impl $name $(<$($generic $(: $($bound)* $bound2)?),+>)?, $ffi_name $(, @type_ $get_type_expr)?);
```

Comment:
For this change I very much need review, because I am not sure my workaround is correct.

The old code did: `$($recurse!($params);)?` 
I assume that it meant: if params are specified then recurse, otherwise ignore and expand to nothing.

Unfortunately this cannot work now, because params contains multiple nested meta variables and it gives:

```
error: meta-variable `gen` repeats 0 times, but `get_type_expr` repeats 1 time
  --> glib/src/boxed.rs:28:10
   |
28 |         $($crate::glib_boxed_wrapper!(@value_impl $name $(<$($gen $(: $bound)?),+>)?, $ffi_name, @type_ $get_type_expr);)?
```
The way I worked it around instead is to make only the the last param `$(, @type_ $get_type_expr)?` optional, and below in the `@value_impl` match the case where `@type_` is not specified and expand to nothing.